File: easybuild-backend/Libraries/Easybuild.DAL/Entities/Contract.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Easybuild.DAL.Entities
{
    public class Contract : IEntity<int>
    {
        public int Id { get; set; }

        public int CustomerId { get; set; }

        public User Customer { get; set; }

        public int PerformerId { get; set; }

        public User Performer { get; set; }

        public ICollection<JobProposal> JobProposals { get; set; }

        public DateTime CreatedDate { get; set; }

        public DateTime? EffectiveDate { get; set; }

        public DateTime ExpirationDate { get; set; }

        public DateTime? SignOffDate { get; set; }

        public ContractStatus Status { get; set; }

        public Guid? ReportId { get; set; }

        [StringLength(256)]
        public string ConstructionSiteAddress { get; set; }

        public Contract()
        {
            JobProposals = new List<JobProposal>();
        }
    }
}

File: easybuild-frontend-web/src/app/shared/models/contract.ts
import { UserIdentity } from './user-identity';
import { JobProposal } from './job-proposal';
import { ContractStatus } from './contract-status';

export class Contract {
    id: number;
    customerId: number;
    customer: UserIdentity;
    performerId: number;
    performer: UserIdentity;
    jobProposals: JobProposal[];
    createdDate: string;
    effectiveDate: string;
    expirationDate: string;
    signOffDate: string;
    reportId: number;
    status: ContractStatus;
    constructionSiteAddress: string;
}